src/main/java/com/es/phoneshop/web/servlets/CheckoutPageServlet.java

```diff
@@ -0,0 +1,136 @@
+import com.es.phoneshop.model.attributesHolder.HttpSessionAttributesHolder;
+import com.es.phoneshop.model.cart.Cart;
+import com.es.phoneshop.model.contactDetails.ContactDetails;
+import com.es.phoneshop.model.deliveryDetails.DeliveryDetails;
+import com.es.phoneshop.model.order.Order;
+import com.es.phoneshop.model.paymentMethod.PaymentMethod;
+import com.es.phoneshop.model.validator.localDateValidator.DeliveryDateValidator;
+import com.es.phoneshop.model.validator.localDateValidator.LocalDateValidator;
+import com.es.phoneshop.model.validator.parameterValidator.NotBlankParameterValidator;
+import com.es.phoneshop.model.validator.parameterValidator.PhoneParameterValidator;
+import com.es.phoneshop.service.cart.CartService;
+import com.es.phoneshop.service.cart.CustomCartService;
+import com.es.phoneshop.service.order.CustomOrderService;
+import com.es.phoneshop.service.order.OrderService;
+import jakarta.servlet.ServletConfig;
+import jakarta.servlet.ServletException;
+import jakarta.servlet.http.HttpServlet;
+import jakarta.servlet.http.HttpServletRequest;
+import jakarta.servlet.http.HttpServletResponse;
+
+import java.io.IOException;
+import java.time.LocalDate;
+import java.time.format.DateTimeParseException;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.function.Consumer;
+
+public class CheckoutPageServlet extends HttpServlet {
+
+    private CartService cartService;
+    private OrderService orderService;
+    private PhoneParameterValidator phoneParameterValidator;
+    private NotBlankParameterValidator notBlankParameterValidator;
+    private LocalDateValidator localDateValidator;
+
+    @Override
+    public void init(ServletConfig config) throws ServletException {
+        super.init(config);
+        cartService = CustomCartService.getInstance();
+        orderService = CustomOrderService.getInstance();
+        phoneParameterValidator = PhoneParameterValidator.getInstance();
+        notBlankParameterValidator = NotBlankParameterValidator.getInstance();
+        localDateValidator = DeliveryDateValidator.getInstance();
+    }
+
+    @Override
+    protected void doGet(HttpServletRequest request, HttpServletResponse response) throws ServletException, IOException {
+        AttributesHolder attributesHolder = new HttpSessionAttributesHolder(request.getSession());
+        Cart cart = cartService.getCart(attributesHolder);
+        if (!cart.getCartItems().isEmpty()) {
+            if (request.getAttribute("order") == null) {
+                Order order = orderService.createOrder(cart);
+                request.setAttribute("order", order);
+            }
+            request.setAttribute("products", cartService.getCartProductsFromAttributesHolder(attributesHolder));
+            request.setAttribute("paymentMethods", PaymentMethod.getMethodNames());
+            request.setAttribute("isEmpty", false);
+        } else {
+            request.setAttribute("isEmpty", true);
```

Comment:
we stored cart in session, so the fact of emptiness of the cartItems can be checked directly in jsp

Reply:
We can check it in jsp, but we have to check cart emptiness in servlet anyway (to avoid creation of unnecessary order object, and no to put unnecessary attributes to request), so we will check cart emptiness twice